tmt/steps/__init__.py

```diff
@@ -91,20 +120,21 @@ def __init__(self, data=None, plan=None, name=None, workdir=None):
                 data['how'] = self.how
             # Ensure that each config has a name
             if 'name' not in data and len(self.data) > 1:
-                raise GeneralError(
+                raise tmt.utils.GeneralError(
                     f"Missing 'name' in the {self} step config "
                     f"of the '{self.plan}' plan.")
 
     @property
-    def enabled(self):
+    def enabled(self) -> Optional[bool]:
         """ True if the step is enabled """
         try:
+            assert self.plan is not None
```

Comment:
I suppose that in original code `self.plan.my_run._context.obj.steps` should raise `AttributeError` here and the method returned `None`. assert is removed and condition was added instead

Reply:
Hmmm, thinking about it a bit more, I believe `plan` should never be `None`. Step instance without an attached plan does not make much sense. So I guess we should make `plan` mandatory in `Step.__init__`. But that would mean moving `plan` parameter before `data`, right?

Reply:
I'm afraid so, yes. You could try switching `Step.__init__()` to use keyword-only arguments, but that might spawn another kind of work, fixing all `Step()` invocations to use keyword parameters. But since `Step.__init__()` is already using no positional parameters, it might also be an empty set of changes.

Reply:
Ok, updated `plan` to be mandatory in 2571784385c4e3c739ab71ff3453cfb06404ac49.